uniprot/uniprot_importer.py

```diff
@@ -0,0 +1,142 @@
+    global uniprot_release
+    while batch_url:
+        response = session.get(batch_url)
+        response.raise_for_status()
+        if uniprot_release is None:
+            uniprot_release = response.headers["X-UniProt-Release"]
+        yield response
+        batch_url = get_next_link(response.headers)
+
+# Function to fetch Uniprot data
+def fetch_all_data():
+    total_items = []
+    for batch in get_batch(url):
+        current_batch_json = batch.json()
+        for item in current_batch_json["results"]:
+            if "comments" in item and len(item["comments"])!=0 and "texts" in item["comments"][0] and len(item["comments"][0]["texts"])!=0 and "value" in item["comments"][0]["texts"][0]:
+                current_item = {}
+                current_item["gene_symbol"] = item["genes"][0]["geneName"]["value"]
+                current_item["accession"] = item["primaryAccession"]
+                current_item["protein_function"] = item["comments"][0]["texts"][0]["value"]
+                current_item["HGNC"] = item["uniProtKBCrossReferences"][0]["id"]
+                current_item["MIM"] = item["uniProtKBCrossReferences"][1]["id"]
+                total_items.append(current_item)
+    print("Uniprot data successfully fetched via Uniprot API.")
+    return total_items
+
+# Function to insert Uniprot data to database
+def insert_uniprot_data(db_host, db_port, db_name, db_user, db_password, total_items):
+    sql_source = f""" SELECT id, name FROM source WHERE name = 'UniProt' or name = 'HGNC'
+                 """
+    sql_identifier = f""" SELECT identifier, locus_id FROM locus_identifier WHERE source_id = %s
+                 """
+    insert_sql = f""" INSERT INTO uniprot_annotation(uniprot_accession, gene_id, hgnc, gene_symbol, mim, protein_function, source_id)
+                  VALUES(%s, %s, %s, %s, %s, %s, %s)
+              """
+    sql_meta = """ INSERT INTO meta(`key`, date_update, is_public, description, source_id, version)
+               VALUES(%s,%s,%s,%s,%s,%s)
+           """
+
+    db = MySQLdb.connect(host=db_host, port=db_port, user=db_user, passwd=db_password, db=db_name)
+    cursor = db.cursor()
+    # Fetch source ids
+    source_ids = {}
+    cursor.execute(sql_source)
+    data = cursor.fetchall()
+    if len(data) != 0:
+        for row in data:
+            source_ids[row[1]] = row[0]
+    # Fetch locus identifiers
+    identifier_to_locus_id_map = {}
+    cursor.execute(sql_identifier, [source_ids['HGNC']])
+    data = cursor.fetchall()
+    if len(data) != 0:
+        for row in data:
+            identifier_to_locus_id_map[row[0]] = row[1]
+    # Insert Uniprot data
+    insert_count = 0
+    for item in total_items:
+        if item["HGNC"] in identifier_to_locus_id_map:
+            cursor.execute(insert_sql, [item["accession"],
+                                  identifier_to_locus_id_map[item["HGNC"]],
+                                  int(item["HGNC"][5:]), # To be changed once column datatype changes
```

Comment:
You could update the include the prefix "HGNC". We are only going to run the import when the db schema is updated to varchar.

Reply:
taken care